### PR TITLE
Reimplemented food so it is unique for each shop

### PR DESCRIPTION
**Pull request by stephlum**
Changed implementation of food so every food item is unique to a shop. As a result, price and availability have been moved to food as attributes + food has a shop id attribute. Menu was also updated to reflect this change and any classes that used food and/or menu.
Also deleted menu item use cases that were redundant to the food use cases.
Updated CartController

### CODE REVIEW

File: src/UseCases/Addon/CreateAddonUseCase.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
      */
     public Boolean createAddon(String vendorToken, String name, float price,
                                ArrayList<Integer> types, boolean availability, String id) {
-        IVendor vendor = vendorRepository.getVendorFromToken(vendorToken);
+        IVendor vendor = (IVendor) vendorRepository.getUserFromToken(vendorToken);
```
Comment (akashilangovan) [inline src/UseCases/Addon/CreateAddonUseCase.java:33]: Why has this been renamed to getUserFromToken? Shouldn't this be getVendorFromToken?

File: src/UseCases/Food/CreateFoodUseCase.java [hunk 2]
```diff
@@ -51,7 +51,7 @@ public CreateFoodUseCase(FoodRepository foodRepo, VendorRepository vendorRepo,
     public boolean createFood(String token, String shopId, String name, String desc,
                               float price, List<ISingleton> singletons){
 
-        IVendor vendor = vendorRepository.getVendorFromToken(token);
+        IVendor vendor = (IVendor)vendorRepository.getUserFromToken(token);
```
Comment (akashilangovan) [inline src/UseCases/Food/CreateFoodUseCase.java:54]: same here

File: src/UseCases/Addon/CreateAddonUseCase.java (hunk 1 above)
Comment (stephlum) [reply src/UseCases/Addon/CreateAddonUseCase.java:33]: Vendor repository extends user repository so we have the getUserFromToken method

File: src/UseCases/Food/CreateFoodUseCase.java (hunk 2 above)
Comment (stephlum) [reply src/UseCases/Food/CreateFoodUseCase.java:54]: Vendor repository extends user repository so we have the getUserFromToken method

File: src/UseCases/Addon/CreateAddonUseCase.java (hunk 1 above)
Comment (akashilangovan) [reply src/UseCases/Addon/CreateAddonUseCase.java:33]: right, makes sense

**Pull request merged**